.buildkite/scripts/publish/publish-common.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```

Comment:
could be worth running shellcheck on this and other bash scripts being added, I think it'll flag some things to fix

Reply:
I didn't know shellcheck was a thing, thanks for the suggestion! I've fixed everything it flagged.